docs/DeveloperGuide.md

```diff
@@ -1,34 +1,251 @@
+
+![Add food sequence diagram](images/AddFoodSequenceDiagram.png)
+
+_Figure 2: Sequence diagram when adding a food into Fitr_
+
+In Figure 2 above, the `Ui` class reads the user's input, which is then parsed by the `Parser` class. Once parsed, it creates a new `AddFoodCommand` class, which is then returned to the `Fitr` class to be executed. The `Food` object is created, then added into `FoodList`. The `FoodList` is then saved to local storage by the `StorageManager` class.
+ 
+### 3.2 Components
+
+This section expands on the different components seen in the architecture section.
+
+#### 3.2.1 Ui component
+
+The `Ui` class handles all user inputs and system output.
+
+This component also listens to other components and outputs the desired messages in specified formats.
+
+#### 3.2.2 Parser component
+
+The Parser component takes in the user input from the `Ui` class and handles the various commands. If the command is invalid, it calls the `Ui` class to prompt the user until valid commands are entered.
+When a valid command is keyed in, the `Parser` class returns a `Command` object to execute the command.
+
+#### 3.2.3 Command component
+
+The Command component consists of an abstract `Command` class and the various different commands that inherit the `Command` class.
+Each type of command class (e.g. `addFoodCommand`, `addExerciseCommand` etc) implements an abstract `execute()` method that carries out the command. 
+
+#### 3.2.4 User component
+
+The `User` class contains all information related to the user data (i.e. `name`, `age`, `height`, `weight`
+and `gender`) and handles all operations on these user data (i.e. `getBmi()`)
+
+The `User` class has two constructors and implementations differ based on the constructor used. 
+
+If a `User` object is instantiated without parameters (i.e. `User user = new User()`), the `setup` method in the
+ `user` object will be invoked, allowing users to go through a set of instructions to input their `name`, `age`, `gender`, `height`, `weight` and
+ `fitnessLevel`.
+ 
+If a `User` object is instantiated with defined parameters (i.e. `User user = new User(name, age, height, weight
+, gender)`), the values from the given parameters will be stored within the user class without going through
+ `setup()`.
+
+On startup, the `Fitr` class instantiates a `Storage` object and calls its `loadUserProfile()` method. This method
+returns a `User` type object and is referenced by a pre-declared `User` type variable, which is used throughout the
+running session. 
+
+#### 3.2.5 Storage component
+
+Under the storage component, the `StorageManager` class handles the read and write operations of the various list classes available, such as `ExerciseList` and `FoodList`, through classes such as `ExerciseStorage` and `FoodStorage`.
+
+![StorageManager class diagram](images/StorageClassDiagram.png)
+
+_Figure 3: `StorageManager` class diagram_
```

Comment:
Again, the diagram seems to insinuate that Goal contains GoalList. If this is by design, perhaps consider renaming Goal to GoalContainer or equivalent to avoid potential confusion. (Same for Food and Exercise).